tests/components/mqtt/test_fan.py

```diff
@@ -291,54 +295,48 @@ async def test_controlling_state_via_topic_no_percentage_topics(hass, mqtt_mock)
     async_fire_mqtt_message(hass, "preset-mode-state-topic", "freaking-high")
     state = hass.states.get("fan.test")
     assert state.attributes.get("preset_mode") == "freaking-high"
-    assert state.attributes.get(fan.ATTR_PERCENTAGE) == 100
+    assert state.attributes.get(fan.ATTR_PERCENTAGE) is None
     # use of speeds is deprecated, support will be removed after a quarter (2021.7)
     assert state.attributes.get("speed") == fan.SPEED_OFF
 
     async_fire_mqtt_message(hass, "preset-mode-state-topic", "high")
```

Comment:
Can we go with something like `auto`, `smart`, `whoosh`, etc so there isn't any future confusion here?

Reply:
`test_fan.py` was updated with `preset_modes` like `auto`, `smart`, `whoosh` to avoid any future confusion.